include/fmt/ranges.h

```diff
@@ -391,56 +420,74 @@ template <typename R>
 using maybe_const_range =
     conditional_t<has_const_begin_end<R>::value, const R, R>;
 
-// is_nonrecursive_range depends on fmt::is_range<T, Char>::value == true.
-// It exists to ensure short-circuit evaluation in the constraint of the 
-// formatter specialization below. A similar approach is used in 
-// https://wg21.link/p2286.
-template <typename R>
-struct is_nonrecursive_range : bool_constant<
-    !std::is_same<uncvref_type<R>, R>::value> {};
-
-// is_formattable_delayed depends on is_nonrecursive_range<R>::value == true.
-// It exists to ensure short-circuit evaluation in the constraint of the 
-// formatter specialization below.
-template <typename R, typename Char>
-struct is_formattable_delayed : disjunction<
-    is_formattable<uncvref_type<maybe_const_range<R>>, Char>,
-    has_fallback_formatter<uncvref_type<maybe_const_range<R>>, Char>> {};
-
 }  // namespace detail
 
-template <typename R, typename Char>
-struct formatter<
-    R, Char,
-    enable_if_t<
-        conjunction<fmt::is_range<R, Char>,
-          detail::is_nonrecursive_range<R>
-// Workaround a bug in MSVC 2015 and earlier.
-#if !FMT_MSC_VERSION || FMT_MSC_VERSION > 1900
-          ,
-          detail::is_formattable_delayed<R, Char>
-#endif
-        >::value
-        >> {
+template <typename T, typename Char, typename Enable = void>
+struct range_formatter;
 
-  using range_type = detail::maybe_const_range<R>;
-  using formatter_type =
-      detail::range_formatter_type<Char, detail::uncvref_type<range_type>>;
-  formatter_type underlying_;
+template <typename T, typename Char>
+struct range_formatter<
+    T, Char,
+    enable_if_t<conjunction<
+        std::is_same<T, remove_cvref_t<T>>,
+        disjunction<is_formattable<T, Char>,
+                    detail::has_fallback_formatter<T, Char>>>::value>> {
+ private:
+  detail::range_formatter_type<Char, T> underlying_;
   bool custom_specs_ = false;
-  bool no_brackets_ = false;
+  basic_string_view<Char> separator_ = FMT_STATICALLY_WIDEN(Char, ", ");
+  basic_string_view<Char> opening_bracket_ = FMT_STATICALLY_WIDEN(Char, "[");
+  basic_string_view<Char> closing_bracket_ = FMT_STATICALLY_WIDEN(Char, "]");
```

Comment:
Let's do what was suggested in https://github.com/fmtlib/fmt/pull/2983#issuecomment-1181291453 and remove `FMT_STATICALLY_WIDEN`.

Reply:
Done, I think.